src/lib.rs

```diff
@@ -758,45 +758,75 @@ fn aom_uleb_size_in_bytes(mut value: u64) -> usize {
+    coded_value[i] = byte;
+  }
+
+  leb_size
+}
+
 #[allow(unused)]
 fn write_obus(packet: &mut Write, sequence: &mut Sequence,
                             fi: &FrameInvariants) -> Result<(), std::io::Error> {
-    let mut uch = BitWriter::<BE>::new(packet);
+    //let mut uch = BitWriter::<BE>::new(packet);
     let obu_extension = 0 as u32;
 
-    uch.write_obu_header(OBU_Type::OBU_TEMPORAL_DELIMITER, obu_extension);
-    uch.write(8,0)?;	// size of payload == 0, one byte
+    let mut buf1 = Vec::new();
+    {
+      let mut bw1 = BitWriter::<BE>::new(&mut buf1);
+      bw1.write_obu_header(OBU_Type::OBU_TEMPORAL_DELIMITER, obu_extension);
+      bw1.write(8,0)?;	// size of payload == 0, one byte
+    }
+    packet.write(&buf1).unwrap();
+    buf1.clear();
 
     // write sequence header obu if KEY_FRAME, preceded by 4-byte size
     if fi.frame_type == FrameType::KEY {
-        let obu_header_size = uch.write_obu_header(OBU_Type::OBU_SEQUENCE_HEADER, obu_extension);
-
-        let obu_payload_error = uch.write_sequence_header_obu(sequence, fi);
-        uch.byte_align()?;
+        let mut buf2 = Vec::new();
+        {
+            let mut bw1 = BitWriter::<BE>::new(&mut buf1);
+            bw1.write_obu_header(OBU_Type::OBU_SEQUENCE_HEADER, obu_extension);
+        }
+        packet.write(&buf1).unwrap();
+        buf1.clear();
 
         let mut obu_payload_size = 0 as u64;
-        match obu_payload_error {
-            Ok(size) => obu_payload_size = size as u64,
-            Err(e) => println!("obu_payload_size error: {:?}", e),
+        {
+            let mut bw2 = BitWriter::<BE>::new(&mut buf2);
+            let error = bw2.write_sequence_header_obu(sequence, fi);
+            bw2.byte_align()?;
+
+            match error {
+                Ok(size) => obu_payload_size = size as u64,
+                Err(e) => println!("obu_payload_size error: {:?}", e),
+            }
+            let mut bw1 = BitWriter::<BE>::new(&mut buf1);
+            // uleb128()
+            let mut coded_payload_length = [0 as u8; 8];
+            let leb_size = aom_uleb_encode(obu_payload_size, &mut coded_payload_length);
+            for i in 0..leb_size {
+                bw1.write(8, coded_payload_length[i]);
+            }
         }
+        packet.write(&buf1).unwrap();
+        buf1.clear();
 
-        // TODO: Looks like we need to write the length (i.e. obu_payload_size)
-        // btw obu_header and obu+payload (i.e. _sequence_header_obu)
-        // So, memmove() seems needed for that.
-
-        let obu_payload_size_uleb = aom_uleb_size_in_bytes(obu_payload_size);
-
-        /*const size_t length_field_size =
-            obu_memmove(obu_header_size, obu_payload_size, data);
-        if (write_uleb_obu_size(obu_header_size, obu_payload_size, data) !=
-            AOM_CODEC_OK) {
-        			return AOM_CODEC_ERROR;
-        }*/
+        packet.write(&buf2).unwrap();
+        buf2.clear();
```

Comment:
@lu-zero : I'd like this checked by you. So, if I call 
packet.write(&buf1).unwrap(); and packet.write(&buf2).unwrap(); in order,
bu1 and bu2 is written to packet buffer in order, correct?

Reply:
I think so.